docs/user-guide/dream/dream-data-reduction.ipynb

```diff
@@ -17,92 +17,70 @@
-    "    UncertaintyBroadcastMode: UncertaintyBroadcastMode.drop,\n",
-    "    # Edges for binning in d-spacing\n",
-    "    DspacingBins: sc.linspace(\"dspacing\", 0.0, 2.3434, 201, unit=\"angstrom\"),\n",
-    "    # Mask in time-of-flight to crop to valid range\n",
-    "    TofMask: lambda x: (x < sc.scalar(0.0, unit=\"ns\"))\n",
-    "    | (x > sc.scalar(86e6, unit=\"ns\")),\n",
-    "    TwoThetaMask: None,\n",
-    "    WavelengthMask: None,\n",
-    "}\n",
-    "\n",
-    "# Not available in simulated data\n",
-    "sample = sc.DataGroup(position=sc.vector([0.0, 0.0, 0.0], unit=\"mm\"))\n",
-    "params[RawSample[SampleRun]] = sample\n",
-    "params[RawSample[VanadiumRun]] = sample\n",
-    "\n",
-    "source = sc.DataGroup(position=sc.vector([-3.478, 0.0, -76550], unit=\"mm\"))\n",
-    "params[RawSource[SampleRun]] = source\n",
-    "params[RawSource[VanadiumRun]] = source\n",
-    "\n",
-    "charge = sc.scalar(1.0, unit=\"µAh\")\n",
-    "params[AccumulatedProtonCharge[SampleRun]] = charge\n",
-    "params[AccumulatedProtonCharge[VanadiumRun]] = charge"
+    "workflow = dream.DreamGeant4Workflow()"
    ]
   },
   {
    "cell_type": "markdown",
-   "id": "21cb87f2-4ff7-436e-b603-cc8f60c73e7a",
+   "id": "1252feab-12d2-46ac-bf74-70b32344473d",
    "metadata": {},
    "source": [
-    "## Create pipeline using Sciline\n",
-    "\n",
-    "We use the `powder` and `geant4` providers to build our pipeline."
+    "We then need to set the missing parameters which are specific to each experiment\n",
+    "(the keys are types defined in [essdiffraction.types](../generated/modules/ess.diffraction.types.rst)):"
    ]
   },
   {
    "cell_type": "code",
    "execution_count": null,
-   "id": "98e99f33-6f4b-4b60-acaf-added3c6e1b0",
+   "id": "502e77cc-0253-4e71-9d97-81ff560ef99d",
    "metadata": {},
    "outputs": [],
    "source": [
-    "providers = (\n",
-    "    *geant4_providers,\n",
-    "    *powder.providers,\n",
-    ")\n",
-    "\n",
-    "pipeline = sciline.Pipeline(providers, params=params)\n",
-    "pipeline = powder.with_pixel_mask_filenames(pipeline, [])"
+    "workflow[Filename[SampleRun]] = dream.data.simulated_diamond_sample()\n",
+    "workflow[Filename[VanadiumRun]] = dream.data.simulated_vanadium_sample()\n",
+    "workflow[Filename[EmptyCanRun]] = dream.data.simulated_empty_can()\n",
+    "workflow[CalibrationFilename] = None\n",
+    "workflow[NeXusDetectorName] = \"mantle\"\n",
+    "# The upper bounds mode is not yet implemented.\n",
+    "workflow[UncertaintyBroadcastMode] = UncertaintyBroadcastMode.drop\n",
+    "# Edges for binning in d-spacing\n",
+    "workflow[DspacingBins] = sc.linspace(\"dspacing\", 0.0, 2.3434, 201, unit=\"angstrom\")\n",
+    "# Mask in time-of-flight to crop to valid range\n",
+    "workflow[TofMask] = lambda x: (x < sc.scalar(0.0, unit=\"ns\")) | (x > sc.scalar(86e6, unit=\"ns\"))\n",
+    "workflow[TwoThetaMask] = None\n",
+    "workflow[WavelengthMask] = None\n",
```

Comment:
Can those be `None` by default?

Reply:
They could, but I thought this would make it obvious what the syntax is if you want to add other masks. I guess this goes back to the discussion on whether these notebooks should only show what is needed, and we need another page that describes everything that can be done in the workflow.